src/cache/diagnostic_edmf_precomputed_quantities.jl

```diff
@@ -338,9 +356,25 @@ NVTX.@annotate function set_diagnostic_edmf_precomputed_quantities_do_integral!(
                 Fields.field_values(Fields.level(ᶠnh_pressure³ʲ, i - 1 - half))
             scale_height =
                 CAP.R_d(params) * CAP.T_surf_ref(params) / CAP.grav(params)
-            S_q_totʲ_prev_level =
-                Fields.field_values(Fields.level(ᶜS_q_totʲ, i - 1))
+
+            if precip_model isa Union{Microphysics0Moment, Microphysics1Moment}
+                S_q_totʲ_prev_level =
+                    Fields.field_values(Fields.level(ᶜS_q_totʲ, i - 1))
+            end
+            if precip_model isa Microphysics1Moment
+                S_q_raiʲ_prev_level =
+                    Fields.field_values(Fields.level(ᶜS_q_raiʲ, i - 1))
+                S_q_snoʲ_prev_level =
+                    Fields.field_values(Fields.level(ᶜS_q_snoʲ, i - 1))
+                S_e_totʲ_prev_level =
+                    Fields.field_values(Fields.level(ᶜS_e_totʲ, i - 1))
+                S_prev_level = Fields.field_values(Fields.level(ᶜSᵖ, i - 1))
```

Comment:
Maybe Sᵖ_prev_level will be more clear? (I don't have strong preference, so up to you.)